tests/src/helpers.rs

```diff
@@ -31,57 +31,45 @@ use jsonrpsee_ws_server::WsServer;
 use std::net::SocketAddr;
 use std::time::Duration;
 
-use futures::channel::oneshot::{Receiver, Sender};
-use futures::future::FutureExt;
+use futures::channel::oneshot::Sender;
 
 pub fn websocket_server(server_started: Sender<SocketAddr>) {
 	std::thread::spawn(move || {
 		let rt = tokio::runtime::Runtime::new().unwrap();
 
-		let server = rt.block_on(WsServer::new("127.0.0.1:0")).unwrap();
-		let mut sub_hello =
-			server.register_subscription("subscribe_hello".to_owned(), "unsubscribe_hello".to_owned()).unwrap();
-		let mut sub_foo =
-			server.register_subscription("subscribe_foo".to_owned(), "unsubscribe_foo".to_owned()).unwrap();
-		let mut call = server.register_method("say_hello".to_owned()).unwrap();
-		server_started.send(*server.local_addr()).unwrap();
+		let mut server = rt.block_on(WsServer::new("127.0.0.1:0")).unwrap();
+		let mut sub_hello = server.register_subscription("subscribe_hello", "unsubscribe_hello").unwrap();
+		let mut sub_foo = server.register_subscription("subscribe_foo", "unsubscribe_foo").unwrap();
+
+		server.register_method("say_hello", |_| Ok("hello")).unwrap();
+
+		server_started.send(server.local_addr().unwrap()).unwrap();
+
+		rt.spawn(server.start());
 
 		rt.block_on(async move {
 			loop {
-				let hello_fut = async {
-					let handle = call.next().await;
-					handle.respond(Ok(JsonValue::String("hello".to_owned()))).await.unwrap();
-				}
-				.fuse();
-
-				let timeout = tokio::time::sleep(Duration::from_millis(100)).fuse();
-				futures::pin_mut!(hello_fut, timeout);
-				futures::select! {
-					_ = hello_fut => (),
-					_ = timeout => {
-						sub_hello.send(JsonValue::String("hello from subscription".to_owned())).await.unwrap();
-						sub_foo.send(JsonValue::Number(1337_u64.into())).await.unwrap();
-					}
-				}
+				tokio::time::sleep(Duration::from_millis(100)).await;
+
+				sub_hello.send(&"hello from subscription").unwrap();
+				sub_foo.send(&1337_u64).unwrap();
 			}
 		});
 	});
 }
 
-pub fn websocket_server_with_wait_period(server_started: Sender<SocketAddr>, wait: Receiver<()>) {
+pub fn websocket_server_with_wait_period(server_started: Sender<SocketAddr>) {
```

Comment:
This doesn't wait for anything anymore though does it? Maybe this should be named `websocket_server` and the other method `websocket_server_with_subs` or something?

Reply:
good catch, I think this change makes the test `ws_more_request_than_buffer_should_not_deadlock` useless because the server would respond to all requests instantly instead of the client has to increase its own buffer to exceed the configured buffer size.

Or am I missing something? :)